src/components/modal/modal.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import closeIcon from "../../assets/closeIcon.svg";
+import { motion, AnimatePresence, useMotionValue } from "framer-motion";
+import useMediaQuery from "../../utils/useMediaQuery";
+/**
+ The **Modal** component is the base of all the modals in the web app.
+ *
+ *  @author [Nuriya](https://github.com/NuriyaAkh)
+ */
+const ModalComponent = styled(motion.div)`
+  box-sizing: border-box;
+  justify-content: center;
+  align-items: center;
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  background-color: rgba(0, 0, 0, 0.5);
+  z-index: 999;
+  transition: visibility 0s ease-in-out 0.4s, opacity 0.4s ease-in-out;
+  margin: 0;
+  padding: 0;
+  color: white;
+  @media (max-width: 1000px) {
+    align-items: flex-end;
+  }
+`;
+
+const ModalWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 375px;
+  min-height: 465px;
+  background: #2d2d41;
+  border-radius: 16px 16px 0px 0px;
+  align-items: center;
+  padding: 16px 16px 24px;
+  box-sizing: border-box;
+  position: relative;
+  justify-content: space-between;
+  transition: visibility 0s ease-in-out 0.4s, opacity 0.4s ease-in-out;
+  @media (min-width: 1000px) {
+    border-radius: 16px 16px 16px 16px;
+  }
+`;
+const ModalBody = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const ModalHeader = styled.div`
+  display: flex;
+  justify-content: center;
+  width: 40px;
+  height: 2px;
+  background: rgba(246, 246, 246, 0.5);
+  border-radius: 1px;
+  margin: 0 auto;
+  cursor: grab;
+  @media (min-width: 1000px) {
+    display: none;
+  }
+`;
+const ModalTitle = styled.h5`
```

Comment:
Bootstrap classes can be used for the text styling too. Some of these match the theme default, so you don't have to specify here too. (Applies to all components)